include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4665,7 +4671,128 @@ struct OrtApi {
+
+  /** \brief Release OrtDevice object.
+   *
+   * \since Version 1.xx.
+   */
+  ORT_CLASS_RELEASE(Device);
+
+  /** \brief Register the plugin ExecutionProvider library
+   *
+   * The plugin ExecutionProvider library will be loaded and EP factory object will be created and saved in OrtEnv object
+   *
+   * \param[in] lib_path the path of the plugin ExecutionProvider library
+   * \param[in] env OrtEnv object
+   * \param[in] ep_name the plugin ExecutionProvider name
+   *
+   * \since Version 1.xx.
+   */
+  ORT_API2_STATUS(RegisterPluginExecutionProviderLibrary, _In_ const ORTCHAR_T* lib_path, _In_ OrtEnv* env, _In_ const char* ep_name);
+
+  /** \brief Unregister the plugin ExecutionProvider library
+   *
+   * The plugin ExecutionProvider factory will be removed from OrtEnv object
+   *
+   * \param[in] env OrtEnv object
+   * \param[in] ep_name the plugin ExecutionProvider name
+   *
+   * \since Version 1.xx.
+   */
+  ORT_API2_STATUS(UnregisterPluginExecutionProviderLibrary, _In_ OrtEnv* env, _In_ const char* ep_name);
+
+  /** \brief Append the plugin ExecutionProvider factory into the session option with provider options
+   *
+   * \param[in] options OrtSessionOptions object
+   * \param[in] ep_name the plugin ExecutionProvider name
+   * \param[in] env OrtEnv object
+   * \param[in] provider_options_keys provider options' keys
+   * \param[in] provider_options_values provider options' values
+   * \param[in] num_keys the number of the provider options' key-value pairs
+   *
+   * \since Version 1.xx.
+   */
+  ORT_API2_STATUS(SessionOptionsAppendPluginExecutionProvider, _In_ OrtSessionOptions* options, _In_ const char* ep_name, _In_ OrtEnv* env,
+                   _In_reads_(num_keys) const char* const* provider_options_keys, _In_reads_(num_keys) const char* const* provider_options_values, _In_ size_t num_keys);
+
+  /** \brief Create OrtTypeConstraints object
+   *
+   * \param[out] OrtTypeConstraints object
+   *
+   * \since Version 1.xx.
+   */
+  ORT_API2_STATUS(CreateOrtTypeConstraints, _Outptr_ OrtTypeConstraints** type_constraints);
+
+  /** \brief Add a specific type constraint into OrtTypeConstraints object
+   *
+   * \param[in] type_constraints OrtTypeConstraints object
+   * \param[in] type_symbol symbol string to represent a specific type
+   * \param[in] type a specific type
+   *
+   * \since Version 1.xx.
+   */
+  ORT_API2_STATUS(AddTypeConstraint, _In_ OrtTypeConstraints* type_constraints, _In_ const char* type_symbol, ONNXTensorElementDataType type);
+
+  /** \brief Release OrtTypeConstraints object.
+   *
+   * \since Version 1.xx.
+   */
+  ORT_CLASS_RELEASE(TypeConstraints);
+
+  /** \brief Create KernelCreateInfo with custom op and type constraints, and register it
+   *
+   * \param[in] kernel_registry Opaque pointer of KernelRegistry object
+   * \param[in] custom_op Custom Op where the kernel compute function is defined
+   * \param[in] type_constraints
```

Comment:
>param[in] type_constraints [](http://example.com/codeflow?start=6&length=26)

Given the fact that type_constraing pointer is not-const, does it mean it gets modified or the ownership is taken by the registry? Please, add this to the documentation.